modules/@apostrophecms/command-menu/index.js

```diff
@@ -0,0 +1,199 @@
+const assert = require('assert').strict;
+const pipe = (...functions) => (initial) => functions.reduce((accumulator, current) => current(accumulator), initial);
+
+module.exports = {
+  options: {
+    alias: 'commandMenu'
+  },
+  handlers(self) {
+    return {
+      'apostrophe:ready': {
+        composeCommands() {
+          self.rawCommands = Object.values(self.apos.modules).flatMap(self.composeCommandsForModule);
+          self.removes = [];
+          self.commands = {};
+          self.groups = {};
+
+          const composed = pipe(self.composeCommand, self.composeRemove, self.composeGroup)({ rawCommands: self.rawCommands });
```

Comment:
I think this strategy might not permit a subclass to add back a "field" (a command) removed by a parent class, because you do all the adds and then all the removes. If you look at the implementation in moog.js it does permit that, because it loops through the steps and applies add, remove and group on each pass. You need equivalent logic.

If it's helpful feel free to factor that logic up to a helper that can also be called from your code.

Also consider whether you can just use a regular cascade rather than an unparsed section, I'm not quite sure if you actually need any differences in the behavior of the logic.  The piece module declares several cascades and has no custom logic to parse them. When you declare a cascade that way it just works like `fields` automatically.